src/aui-form-builder/js/aui-form-builder-pages.js

```diff
@@ -495,6 +504,24 @@ A.FormBuilderPages = A.Base.create('form-builder-pages', A.Base, [], {
         this._uiSetActivePageNumber(activePageNumber);
     },
 
+    _uiSetPaginationType: function(type) {
+        var activePageNumber = this.get('activePageNumber'),
+            pagination = this._getPagination(),
+            tabview = this._getTabView();
+
+        if (type === 'tabs') {
+            pagination.get('contentBox').hide();
+            tabview.get('contentBox').show();
+        }
+        else {
+            pagination.get('contentBox').show();
+            tabview.get('contentBox').hide();
+        }
+
+        pagination.set('page', activePageNumber);
+        tabview.selectChild(activePageNumber - 1);
```

Comment:
It's necessary to keep the user at the same page after the change of the pagination type.